code/game/objects/objs.dm

```diff
@@ -18,6 +18,8 @@ var/global/list/reagents_to_log = list(FUEL, PLASMA, PACID, SACID, AMUTATIONTOXI
 	var/damtype = "brute"
 	var/force = 0
 
+	var/w_class
```

Comment:
So that `obj` can take damage when being propelled and colliding with something, with damage scaling with the `w_class` of the object. Idea being that with a toaster and a pen both hitting a wall at the same speed, the toaster would take more damage. In most situations it does seem that non-item objects are generally only moved by players pushing and pulling them, but there are some situations where they can or should be able to be propelled; pressure gradients, singularities, explosions, and possibly others. Mechs are one example of a non-item `obj` with `throw_at()`. Down the line a more robust momentum/impulse system could be developed, but for now I'm using `w_class` as a surrogate for the mass of the object, which `item/kick_act()` was already doing.

Reply:
works for me